### PR TITLE
Enable source-build pre-built detection

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
Fixes: https://github.com/dotnet/deployment-tools/issues/264

### CODE REVIEW

File: eng/SourceBuildPrebuiltBaseline.xml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 <UsageData>
   <IgnorePatterns>
-    <UsagePattern IdentityGlob="*/*" />
+    <UsagePattern IdentityGlob="Microsoft.SourceBuild.Intermediate.*/*" />
+
+    <!-- Ignore System.Text.JSon 7.0.1 and dependencies until we can add
```
Comment (MichaelSimons) [inline eng/SourceBuildPrebuiltBaseline.xml:5]: Would be IMO good to open a tracking issue for this and add it to the comment prior to merging.

Comment (NikolaMilosavljevic) [reply eng/SourceBuildPrebuiltBaseline.xml:5]: > Would be IMO good to open a tracking issue for this and add it to the comment prior to merging.

Yeah, meant to do that originally - working on it now.

Comment (NikolaMilosavljevic) [reply eng/SourceBuildPrebuiltBaseline.xml:5]: Fixed in https://github.com/dotnet/deployment-tools/pull/266/commits/eaff4f995a881cd6442de7dfaf93e86340482fe3

**Review by MichaelSimons — APPROVED** [review]

File: eng/SourceBuildPrebuiltBaseline.xml [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+         7.0 targeting pack and these packages to SBRP.
+         https://github.com/dotnet/source-build/issues/3330 -->
+    <UsagePattern IdentityGlob="System.Text.Json/7.0.1" />
+    <UsagePattern IdentityGlob="Microsoft.Bcl.AsyncInterfaces/7.0.0" />
+    <UsagePattern IdentityGlob="System.Text.Encodings.Web/7.0.0" />
```
Comment (ViktorHofer) [inline eng/SourceBuildPrebuiltBaseline.xml:10]: Curious, why can't those be live references from runtime instead?

Comment (mmitche) [reply eng/SourceBuildPrebuiltBaseline.xml:10]: Because we should not introduce more places where we need to worry about coherency :)

Comment (ViktorHofer) [reply eng/SourceBuildPrebuiltBaseline.xml:10]: I thought that live dependencies are always preferred over SBRP (for obvious reasons). deployment-tools is part of the VMR. Is another node repo node in the graph that bad?

Comment (mmitche) [reply eng/SourceBuildPrebuiltBaseline.xml:10]: It's bad in context of the current build. That means every time we have a servicing patch, preview, etc. We have yet another parallel path to flow runtime through before we can release the product. We only really need to depend on live dependencies where the consuming repo repackages meaningful info about the dependency. For instance, the SDK will repackage the actual binaries, so it needs to depend on the latest. Consider roslyn, it targets net6 in it's build, but inserts into the SDK. It rolls forward to run on the SDK runtime.

**Pull request merged**